components/director/internal/domain/bundle/service.go

```diff
@@ -201,11 +204,54 @@ func (s *service) ListByApplicationIDs(ctx context.Context, applicationIDs []str
 		return nil, err
 	}
 
+	consumerInfo, err := consumer.LoadFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	if pageSize < 1 || pageSize > 200 {
 		return nil, apperrors.NewInvalidDataError("page size must be between 1 and 200")
 	}
 
-	return s.bndlRepo.ListByApplicationIDs(ctx, tnt, applicationIDs, pageSize, cursor)
+	bundlePages, err := s.bndlRepo.ListByApplicationIDs(ctx, tnt, applicationIDs, pageSize, cursor)
+	if err != nil {
+		return nil, err
+	}
+
+	// override the default instance auth only for runtime consumers
+	if consumerInfo.ConsumerType != consumer.Runtime {
+		return bundlePages, nil
+	}
+
+	bundleInstanceAuths, err := s.biaSvc.ListByRuntimeID(ctx, consumerInfo.ConsumerID)
+	if err != nil {
+		return nil, err
+	}
+
+	bundlesCount := 0
+	for _, page := range bundlePages {
+		bundlesCount += len(page.Data)
+	}
+
+	bundleIDToBundleInstanceAuths := make(map[string][]*model.BundleInstanceAuth, bundlesCount)
+	for i, auth := range bundleInstanceAuths {
```

Comment:
If the bundlesCount above is zero and in the same time we have non-empty `bundleInstanceAuths` we'll go through them, build the map and then it won't be used because the for below(`for _, page := range bundlePages`) and the `bundlePages` will be empty?

Seems this for will be obsolete in case of empty bundlePages/bundleCount? Is that possible? We could check if the bundleCounts is 0 to return the bundlePages and do nothing? If it's different than 0 then we'll have the current logic?

Reply:
It should not be possible to have bundle instance auths but no bundles as the bundle instance auths have foreign key for the bundles